src/main/java/org/jenkinsci/plugins/badge/StatusImage.java

```diff
@@ -41,7 +41,7 @@
     private final String length;
 
     StatusImage(String fileName) throws IOException {
-        etag = Jenkins.RESOURCE_PATH+'/'+fileName;
+        etag = '"' + Jenkins.RESOURCE_PATH + '/' + fileName + '"';
```

Comment:
Well, ETag is supposed to be a collision-resistant resource contents hash. For fixed status images the hash can be precomputed, for others it has to be computed on the fly. Resource URI is definitely not what HTTP spec envisions, though newer JDK String.hashCode() will probably work most of the time, assuming resources are always distinctly unique for each URI. Otherwise a hash of contents has to be generated on the fly.